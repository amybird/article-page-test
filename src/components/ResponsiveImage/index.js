import React from 'react';
import PropTypes from 'prop-types';
import { breakpoints } from '../../styles/mediaQueries';

const ResponsiveImage = ({ smallImg, largeImg, imgAlt }) => (
  <picture>
    {largeImg && <source srcSet={largeImg} media={`(min-width: ${breakpoints.medium}px)`} />}
    <img alt={imgAlt} src={smallImg} />
  </picture>
);

ResponsiveImage.propTypes = {
  smallImg: PropTypes.string.isRequired,
  largeImg: PropTypes.string,
  imgAlt: PropTypes.string
};

ResponsiveImage.defaultProps = {
  largeImg: null,
  imgAlt: 'this is a lovely image'
};

export default ResponsiveImage;
