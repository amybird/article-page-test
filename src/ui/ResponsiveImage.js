import React from 'react';
import PropTypes from 'prop-types';
import { breakpoints } from '../styles/mediaQueries';

const ResponsiveImage = ({smallImg, largeImg, imgAlt}) => (
  <picture>
    {largeImg && <source srcSet={largeImg} media={`(min-width: ${breakpoints.medium}px)`} />}
    <img alt={imgAlt} src={smallImg}/>
  </picture>
);

ResponsiveImage.propTypes = {
    smallImg: PropTypes.string.isRequired
};

export default ResponsiveImage;