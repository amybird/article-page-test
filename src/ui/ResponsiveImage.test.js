import React from 'react';
import ResponsiveImage from './ResponsiveImage.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';

it('renders without source tag for smallImg only', () => {
  const tree = renderer
    .create(<ResponsiveImage smallImg="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with source tag when largeImg', () => {  
  const ResponsiveImageComponent = mount(<ResponsiveImage smallImg="test" largeImg="sddsFS" />);
     expect(ResponsiveImageComponent.find('source'));
});
