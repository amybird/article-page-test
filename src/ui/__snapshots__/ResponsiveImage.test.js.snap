// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without source tag for smallImg only 1`] = `
<picture>
  <img
    src="test"
  />
</picture>
`;
